### PR TITLE
feat: shift from cargo test to cargo nextest

### PR DESCRIPTION
**Pull request by Arya-A-Nair**
Shifting from cargo test to cargo nextest as per Issue- [#999](https://github.com/xmtp/libxmtp/issues/999)

### CODE REVIEW

**Comment by insipx** [comment]
Hello!

Thanks for your external contribution!

I approved the workflows, and it seems the github workflow files need a bit more work:

```
error: no such command: `nextest`
	Did you mean `test`?
	View all installed commands with `cargo --list`
	Find a package to install `nextest` with `cargo search cargo-nextest`
```

**Review by insipx — CHANGES_REQUESTED** [review]
CI

File: Dockerfile
Comment (mkysel) [inline Dockerfile:11]: @insipx are we using the dev container?

Comment (insipx) [reply Dockerfile:11]: I don't use it myself, but unsure if anyone else does

Comment (Arya-A-Nair) [reply Dockerfile:11]: The readme file stated to use the vscode dev container, even when I had to run the nextest command locally I had to install it, so for easier developer access I included it in the docker file

**Comment by Arya-A-Nair** [comment]
> Hello!
> 
> Thanks for your contribution!
> 
> I approved the workflows, and it seems the github workflow files need a bit [more work](https://nexte.st/docs/installation/pre-built-binaries/#using-nextest-in-github-actions):
> 
> ```
> error: no such command: `nextest`
> 	Did you mean `test`?
> 	View all installed commands with `cargo --list`
> 	Find a package to install `nextest` with `cargo search cargo-nextest`
> ```

I had added the installation for cargo nextest in the docker file, but forgot to put the same in workflow, I have corrected the mistake in the latest commit

**Comment by mkysel** [comment]
I would use https://github.com/taiki-e/install-action to install the tooling. The nextest examples themselves use it https://github.com/nextest-rs/reuse-build-partition-example/blob/main/.github%2Fworkflows%2Fci.yml

**Comment by Arya-A-Nair** [comment]
> I would use https://github.com/taiki-e/install-action to install the tooling. The nextest examples themselves use it https://github.com/nextest-rs/reuse-build-partition-example/blob/main/.github%2Fworkflows%2Fci.yml

Hey @mkysel , I have replaced that particular piece of code in the latest commit

**Comment by mkysel** [comment]
Thanks @Arya-A-Nair. Looks good to me. I will wait for @insipx to approve since he is the original reporter.

**Review by insipx — APPROVED** [review]
Looks great! Thank you

**Comment by Arya-A-Nair** [comment]
> Looks great, thanks again for the contribution!
> 
> feel free to merge whenever

Hey, as I don't have write access I cant merge from my end

**Comment by mkysel** [comment]
Done!

**Pull request merged**
